### PR TITLE
front: stdcm: add linked path search

### LINKED ISSUE #8593: Search for a linked path in the stdcm interface
### Description and goal

Implementation of the "reference path" search component in the new stdcm interface.

### Acceptance criteria

- [ ] The user can select a reference path after a search in the BASIC grid.
- [ ] Information about the selected path is included in the simulation sheet.

### PR DESCRIPTION
**Pull request by SarahBellaha**
Closes #8593

### CODE REVIEW

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  const selectableSlot = useMemo(() => {
+    const startDate = searchDatetimeWindow ? new Date(searchDatetimeWindow.begin) : new Date();
+    startDate.setUTCHours(0, 0, 0);
+    return {
+      start: startDate,
+      end: searchDatetimeWindow?.end || startDate,
+    };
+  }, [searchDatetimeWindow]);
+
+  const [linkedPathDate, setLinkedPathDate] = useState<Date>(selectableSlot.start);
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: You can rely on TS for the typing
```suggestion
  const [linkedPathDate, setLinkedPathDate] = useState(selectableSlot.start);
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <StdcmCard
+          name={cardName}
+          title={
+            <button type="button" onClick={() => setShowLinkedPathSearch(false)}>
+              {t('translation:common.delete').toLowerCase()}
+            </button>
+          }
+          className="linked-path"
+        >
+          <div className="d-flex flex-column">
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]: You can remove this div if you remove the css property `display: flex` in the className stdcm-card__body
```suggestion
```

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      );
+
+      if (!filteredResults.length) {
+        setDisplaySearchButton(true);
+        return;
+      }
+
+      const newLinkedPathResults = await Promise.all(
+        filteredResults.map(async (result) => {
+          const opList = result.path as PathItemOperationalPoint[];
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: This cast is dangerous, we should check that the path steps are really operationalPoints and return if it's not the case

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        query: [
+          'and',
+          ['search', ['train_name'], linkedPathID],
+          ['=', ['timetable_id'], timetableID!],
+        ],
+      }) as SearchPayload,
+    [timetableID, linkedPathID]
+  );
+
+  const operationalPointSearch = useCallback(
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: Maybe `getExtremitiesDetails` is more precise
```suggestion
  const getExtremitiesDetails = useCallback(
```

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        filteredResults.map(async (result) => {
+          const opList = result.path as PathItemOperationalPoint[];
+          const opIds = {
+            originID: opList.at(0)!.operational_point,
+            destinationID: opList.at(-1)!.operational_point,
+          };
+          const opDetails = (await operationalPointSearch(opIds)) as {
+            origin: SearchResultItemOperationalPoint;
+            destination: SearchResultItemOperationalPoint;
+          };
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: You can move some logic in getExtremitiesDetails 
```suggestion
          const opDetails = await getExtremitiesDetails(result.path);
```

File: front/src/applications/stdcm/types.ts [hunk 6]
```diff
@@ -125,3 +145,15 @@ export enum StdcmStopTypes {
   DRIVER_SWITCH = 'driverSwitch',
   SERVICE_STOP = 'serviceStop',
 }
+
+export type StdcmLinkedPathStep = {
+  date: string;
+  time: string;
+  isoArrivalTime: string;
+} & SearchResultItemOperationalPoint;
```
Comment (clarani) [inline front/src/applications/stdcm/types.ts]: Give only the fields you will need
```suggestion
  trigram: string;
};
```

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+      }
+
+      const newLinkedPathResults = await Promise.all(
+        filteredResults.map(async (result) => {
+          const opList = result.path as PathItemOperationalPoint[];
+          const opIds = {
+            originID: opList.at(0)!.operational_point,
+            destinationID: opList.at(-1)!.operational_point,
+          };
+          const opDetails = (await operationalPointSearch(opIds)) as {
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: - remove the cast and in the function, return null if there was an error or if one of the op does not have a operational point
- if opDetails is undefined, we should return null or undefined here too (you may need to add a compact after that)

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      }
+    },
+    [postSearch]
+  );
+
+  const launchTrainScheduleSearch = useCallback(async () => {
+    setDisplaySearchButton(false);
+    try {
+      const results = (await postSearch({
+        searchPayload,
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: I think you can remove the useMemo searchPayload and directly move the code here (you will need to add `linkedPathID` and `timetableID` to the dependencies of the hook) : 
```suggestion
        searchPayload: {
          object: 'trainschedule',
          query: [
            'and',
            ['search', ['train_name'], linkedPathID],
            ['=', ['timetable_id'], timetableID!],
          ],
        },
```

File: front/src/utils/date.ts [hunk 9]
```diff
@@ -242,3 +249,33 @@ export const formatDateString = (date?: Date | null) => {
   if (!date) return '';
   return dayjs(date).format('DD/MM/YY');
 };
+
+export const formatLocaleDate = (date: Date) => date.toLocaleString().substring(0, 10);
+
+export const isEqualDate = (searchDate: Date, startDate: Date) =>
+  formatLocaleDate(searchDate) === formatLocaleDate(startDate);
+
+export const computeOpSchedules = (startTime: string, secondsFromStartTime: string) => {
```
Comment (clarani) [inline front/src/utils/date.ts]: Can you move this function in `stdcm/utils` ? It's not really a general utils

File: front/src/utils/timeManipulation.ts
Comment (clarani) [inline front/src/utils/timeManipulation.ts:1]: I think you can remove this function

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+const useLinkedPathSearch = () => {
+  const dispatch = useAppDispatch();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const { getTimetableID, getSearchDatetimeWindow } = useOsrdConfSelectors();
+  const { updateOrigin, updateDestination, updateAnteriorPath, updatePosteriorPath } =
+    useOsrdConfActions() as StdcmConfSliceActions;
+
+  const timetableID = useSelector(getTimetableID);
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: Do you think you can use `Id` instead of `ID` ?

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+    startDate.setUTCHours(0, 0, 0);
+    return {
+      start: startDate,
+      end: searchDatetimeWindow?.end || startDate,
+    };
+  }, [searchDatetimeWindow]);
+
+  const [linkedPathDate, setLinkedPathDate] = useState<Date>(selectableSlot.start);
+  const [displaySearchButton, setDisplaySearchButton] = useState(true);
+  const [linkedPathID, setLinkedPathID] = useState<string>();
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: - remove the type string and initialize the string to ''
- rename the props trainNameInput
```suggestion
  const [trainNameInput, setTrainNameInput] = useState('');
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+            <div className="d-flex pr-1 pl-3">
+              <Input
+                id="linked-path-ID"
+                type="text"
+                value={linkedPathID || ''}
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]: If linkedPathID (or trainNameInput) has been initialized, you can remove the `|| ''`
```suggestion
                value={linkedPathID}
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+                onChange={(e) => {
+                  setDisplaySearchButton(true);
+                  setLinkedPathID(e.target.value);
+                }}
+                label="N°"
+              />
+              <DatePicker
+                inputProps={{
+                  id: `linked-path-date`,
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]:
```suggestion
                  id: 'linked-path-date',
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+                  label: 'Date',
+                  name: 'op-date',
+                  onChange: () => {},
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]:
```suggestion
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+                }}
+                selectableSlot={selectableSlot}
+                value={linkedPathDate}
+                onDateChange={(e) => {
+                  setLinkedPathDate(e);
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]:
```suggestion
                onDateChange={(date) => {
                  setLinkedPathDate(date);
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathResults.tsx [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+  onLinkedPathSelection,
+  linkedOp,
+}: StdcmLinkedPathResultsProps) => (
+  <div className="stdcm-linked-path-result">
+    {linkedPathResults.map(({ trainID, origin, destination }, index) => (
+      <button
+        key={`linked-path-${index}`}
+        tabIndex={0}
+        type="button"
+        className="d-flex flex-column"
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathResults.tsx]: You can remove this className
```suggestion
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathResults.tsx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+import type { StdcmLinkedPathResult, StdcmLinkedPathStep } from '../../types';
+
+type StdcmLinkedPathResultsProps = {
+  linkedPathResults: StdcmLinkedPathResult[];
+  onLinkedPathSelection: (
+    pathStepToUpdate: ExtremityPathStepType,
+    trainID: string,
+    linkedPathStep: StdcmLinkedPathStep
+  ) => void;
+  linkedOp: ExtremityPathStepType;
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathResults.tsx]:
```suggestion
  extremityType: ExtremityPathStepType;
```

File: front/src/applications/stdcm/types.ts [hunk 18]
```diff
@@ -125,3 +145,15 @@ export enum StdcmStopTypes {
...
+
+export type StdcmLinkedPathResult = {
+  trainID: string;
```
Comment (clarani) [inline front/src/applications/stdcm/types.ts]:
```suggestion
  trainName: string;
```

File: front/src/applications/stdcm/types.ts [hunk 19]
```diff
@@ -54,6 +56,23 @@ export type StdcmResults = {
...
+        trainID: string;
+      }
+    | undefined;
+  posteriorPath:
+    | {
+        date: string;
+        time: string;
+        trainID: string;
+      }
+    | undefined;
```
Comment (clarani) [inline front/src/applications/stdcm/types.ts]:
```suggestion
  anteriorPath?: {
    date: string;
    time: string;
    trainName: string;
  };
  posteriorPath?: {
    date: string;
    time: string;
    trainName: string;
  };
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 20]
```diff
@@ -0,0 +1,92 @@
...
+      display: flex;
+      flex-direction: column;
+      width: 100%;
+      margin-bottom: 7px;
+    }
+
+    .linked-path-result-infos:not(:first-child) {
+      padding-top: 12px;
+      border-top: 1px solid var(--grey20);
+    }
```
Comment (Uriel-Sautron) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]:
```suggestion
    .linked-path-result-infos {
      display: flex;
      flex-direction: column;
      width: 100%;
      margin-bottom: 7px;
      
      &:not(:first-child) {
        padding-top: 12px;
        border-top: 1px solid var(--grey20);
      }
    }
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+      margin-bottom: 4px;
+    }
+    .opDetails {
+      font-size: 14px;
+      line-height: 20px;
+      margin-bottom: 4px;
+    }
+    .opDetails:not(:last-child) {
+      margin-right: 8px;
+    }
```
Comment (Uriel-Sautron) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]:
```suggestion
    .opDetails {
      font-size: 14px;
      line-height: 20px;
      margin-bottom: 4px;
    
      &:not(:last-child) {
        margin-right: 8px;
      }
    }
```

File: front/src/styles/scss/applications/stdcm/_card.scss [hunk 22]
```diff
@@ -88,97 +88,104 @@
...
         display: grid;
         grid-template-columns: 7.4375rem 7.938rem;
         justify-content: space-between;
-        padding-block: 0.6875rem;
+        padding-block: 8px;
 
         .osrd-config-item {
           padding-right: 0.188rem;
         }
+        .weight {
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_card.scss:140]: I don't see the classes weight and length in the html, is it normal ? Why did you add them ?

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+  color: var(--grey50);
+}
+
+.grey80 {
+  color: var(--grey80);
+}
+
+.info60 {
+  color: var(--info60);
+}
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: If we really want to define these general classes, it should not be here (these classes can be used anywhere in the application since they're not encapsulated...)

I think it's not a good practice, we should move them in the class opDetails

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  .stdcm-card__body {
+    display: flex;
+    width: 450px;
+
+    &.add-linked-path,
+    &.linked-path {
+      padding: 11px 24px 13px 12px;
+    }
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: It would be better to move this class directly in `_card.scss`, next to all the other styles of `stdcm-card__body`

File: front/src/common/types.ts [hunk 25]
```diff
@@ -35,3 +35,5 @@ export type RangedValue = {
   end: number;
   value: string;
 };
+
+export type ExtremityPathStepType = 'origin' | 'destination';
```
Comment (clarani) [inline front/src/common/types.ts]: Since this type is only used in stdcm right now, you can move it in `stdcm/types.ts`

File: front/src/styles/scss/applications/stdcm/_card.scss
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_card.scss:1]: Can you separate the conversion rem to px in another commit ? I can't see if you have modified a value :/

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+.grey80 {
+  color: var(--grey80);
+}
+
+.info60 {
+  color: var(--info60);
+}
+
+.stdcm-linked-path-search-container {
+  max-width: 450px;
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: You can remove this props I think
```suggestion
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 27]
```diff
@@ -0,0 +1,118 @@
+/* eslint-disable no-console */
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]:
```suggestion
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx [hunk 28]
```diff
@@ -0,0 +1,118 @@
...
+            <StdcmLinkedPathResults
+              linkedPathResults={linkedPathResults}
+              onLinkedPathSelection={updateLinkedPathStep}
+              extremityType={linkedOp}
+            />
+          ) : (
+            hasSearchBeenLaunched && (
+              <p className="text-center mb-0">{t('noCorrespondingResults')}</p>
+            )
+          )}
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]: If you define linkedPathResults like this (undefined by default): 
```
const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>();
```
then you can do:
```suggestion
          {linkedPathResults && (
            <>
              {linkedPathResults.length > 0 ? (
                <StdcmLinkedPathResults
                  linkedPathResults={linkedPathResults}
                  onLinkedPathSelection={updateLinkedPathStep}
                  linkedOp={linkedOp}
                />
              ) : (
                hasSearchBeenLaunched && (
                  <p className="text-center mb-0">{t('noCorrespondingResults')}</p>
                )
              )}
            </>
          )}
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    .opDetails {
+      font-size: 14px;
+      line-height: 20px;
+      margin-bottom: 4px;
+
+      &:not(:last-child) {
+        margin-right: 8px;
+      }
+    }
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: You can move the l.75-88 in .linked-path-result-infos

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 30]
```diff
@@ -0,0 +1,90 @@
...
+  .stdcm-card__body {
+    width: 450px;
+
+    &.add-linked-path,
+    &.linked-path {
+      padding: 11px 8px 13px;
+    }
+  }
+
+  .stdcm-linked-path-button {
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: You can move the class stdcm-linked-path-button in linked-path (you can add a class .linked-path)

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 31]
```diff
@@ -0,0 +1,90 @@
...
+    background-color: var(--grey10);
+    color: var(--primary50);
+    border-radius: 6px;
+    margin: 21px 0 13px 15px;
+    &.white {
+      background-color: var(--white);
+    }
+  }
+
+  .stdcm-linked-path-loading {
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]: You can move this class in `stdcm-linked-path-button`

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 32]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  }, [postSearch, trainNameInput, timetableId, linkedPathDate, getExtremitiesDetails]);
+
+  const updateLinkedPathStep = useCallback(
+    (linkedPathStep: ExtremityPathStepType, trainName: string, pathStep: StdcmLinkedPathStep) => {
+      const { name, ch, uic, geographic, isoArrivalTime, date, time, trigram } = pathStep;
+      const newPathStep = {
+        name,
+        ch,
+        id: nextId(),
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: You will need to give the correct id here now, because of my PR :/ 
You should get the stdcmOrigin and stdcmDestination at the top of the file, and then:
```suggestion
        id: linkedPathStep === 'origin' ? origin.id : destination.id,
```
You may need to add more props

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 33]
```diff
@@ -0,0 +1,163 @@
...
+
+import { ArrivalTimeTypes, type StdcmLinkedPathResult, type StdcmLinkedPathStep } from '../types';
+import computeOpSchedules from '../utils/computeOpSchedules';
+
+const useLinkedPathSearch = () => {
+  const dispatch = useAppDispatch();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  const { getTimetableID, getSearchDatetimeWindow } = useOsrdConfSelectors();
+  const { updateOrigin, updateDestination, updateAnteriorPath, updatePosteriorPath } =
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: You need to use the new action `updateStdcmPathStep`
```suggestion
  const { updateStdcmPathStep, updateAnteriorPath, updatePosteriorPath } =
```

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 34]
```diff
@@ -0,0 +1,163 @@
...
+      );
+      setLinkedPathResults(compact(newLinkedPathResults));
+      setHasSearchBeenLaunched(true);
+    } catch (error) {
+      console.error('Train schedule search failed:', error);
+      setDisplaySearchButton(true);
+    }
+  }, [postSearch, trainNameInput, timetableId, linkedPathDate, getExtremitiesDetails]);
+
+  const updateLinkedPathStep = useCallback(
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]: Maybe actually we should transform this function in an action ? And move all the following code in the osrdConf/index.ts ?

File: front/src/applications/stdcm/components/StdcmForm/StdcmOrigin.tsx [hunk 35]
```diff
@@ -33,8 +33,14 @@ const StdcmOrigin = ({
...
-        arrivalToleranceBefore: origin?.arrivalToleranceBefore || DEFAULT_TOLERANCE,
-        arrivalToleranceAfter: origin?.arrivalToleranceAfter || DEFAULT_TOLERANCE,
+        arrivalToleranceBefore:
+          origin?.arrivalToleranceBefore === undefined
+            ? DEFAULT_TOLERANCE
+            : origin.arrivalToleranceBefore,
+        arrivalToleranceAfter:
+          origin?.arrivalToleranceAfter === undefined
+            ? DEFAULT_TOLERANCE
+            : origin.arrivalToleranceAfter,
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmOrigin.tsx]: Why these changes ? I think the previous version does the same thing and is more concise

File: front/src/applications/stdcm/components/StdcmForm/StdcmInputVia.tsx [hunk 36]
```diff
@@ -45,7 +45,7 @@ const StdcmInputVia = ({ stopType, stopDuration, updatePathStepStopTime }: Stdcm
 
   return (
     stopType !== StdcmStopTypes.PASSAGE_TIME && (
-      <div className="stdcm-v2-via-stop-for stop-time">
+      <div className="stdcm-via-stop-for stop-time">
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmInputVia.tsx]: I'm not sure that this class is useful, I can't find it in the css
```suggestion
      <div className="stop-time">
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmDestination.tsx [hunk 37]
```diff
@@ -39,8 +39,14 @@ const StdcmDestination = ({
...
-        arrivalToleranceBefore: destination?.arrivalToleranceBefore || DEFAULT_TOLERANCE,
-        arrivalToleranceAfter: destination?.arrivalToleranceAfter || DEFAULT_TOLERANCE,
+        arrivalToleranceBefore:
+          destination?.arrivalToleranceBefore === undefined
+            ? DEFAULT_TOLERANCE
+            : destination.arrivalToleranceBefore,
+        arrivalToleranceAfter:
+          destination?.arrivalToleranceAfter === undefined
+            ? DEFAULT_TOLERANCE
+            : destination.arrivalToleranceAfter,
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmDestination.tsx]: Same here, the behaviour is not the same ?

(nit: it could be moved to a new commit)

File: front/src/reducers/osrdconf/types.ts [hunk 38]
```diff
@@ -49,6 +54,7 @@ export interface OsrdStdcmConfState extends OsrdConfState {
   totalMass?: number;
   totalLength?: number;
   maxSpeed?: number;
+  linkedPaths: LinkedPaths;
```
Comment (clarani) [inline front/src/reducers/osrdconf/types.ts:59]: You should move the this in `StdcmPathStep` (where `isVia` is false)

File: front/src/styles/scss/applications/stdcm/_card.scss (hunk 22 above)
Comment (SarahBellaha) [reply front/src/styles/scss/applications/stdcm/_card.scss:140]: Oh, they were added in a previous rebase. I will remove them if they aren't used.

File: front/src/applications/stdcm/hooks/useLinkedPathSearch.ts [hunk 39]
```diff
@@ -0,0 +1,170 @@
...
+    startDate.setUTCHours(0, 0, 0);
+    return {
+      start: startDate,
+      end: searchDatetimeWindow?.end || startDate,
+    };
+  }, [searchDatetimeWindow]);
+
+  const [displaySearchButton, setDisplaySearchButton] = useState(true);
+  const [hasSearchBeenLaunched, setHasSearchBeenLaunched] = useState(false);
+  const [trainNameInput, setTrainNameInput] = useState<string>('');
```
Comment (clarani) [inline front/src/applications/stdcm/hooks/useLinkedPathSearch.ts]:
```suggestion
  const [trainNameInput, setTrainNameInput] = useState('');
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 40]
```diff
@@ -0,0 +1,88 @@
...
+    margin-bottom: 18px;
+  }
+
+  &.posterior-linked-path {
+    margin-top: 18px;
+  }
+
+  .stdcm-card__header {
+    button {
+      font-size: 14px;
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]:
```suggestion
      font-size: .875rem;
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 41]
```diff
@@ -0,0 +1,88 @@
...
+      width: 100%;
+      margin-bottom: 7px;
+
+      &:not(:first-child) {
+        padding-top: 12px;
+        border-top: 1px solid var(--grey20);
+      }
+
+      .train-name {
+        font-size: 16px;
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]:
```suggestion
        font-size: 1rem;
```

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss [hunk 42]
```diff
@@ -0,0 +1,88 @@
...
+        padding-top: 4px;
+        margin-bottom: 4px;
+      }
+      .opDetails {
+        font-size: 14px;
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_linkedPath.scss]:
```suggestion
        font-size: .875rem;
```

File: front/src/styles/scss/applications/stdcm/_card.scss [hunk 43]
```diff
@@ -204,7 +211,7 @@
     }
     button {
       display: contents;
-      font-size: 1rem;
+      font-size: 16px;
```
Comment (clarani) [inline front/src/styles/scss/applications/stdcm/_card.scss]:
```suggestion
      font-size: 1rem;
```

File: front/src/applications/stdcm/components/StdcmForm/StdcmConfig.tsx [hunk 44]
```diff
@@ -139,35 +141,47 @@ const StdcmConfig = ({
...
-                  removeDestinationArrivalTime={removeDestinationArrivalTime}
-                />
-              )}
+        <div className="d-flex flex-column">
+          <StdcmLinkedPathSearch
+            defaultCardText={t('indicateAnteriorPath')}
+            cardName={t('trainPath.anteriorPath')}
+            cardIcon={<ArrowUp size="lg" />}
+            className="anterior-linked-path"
+            linkedOp={{ extremityType: 'destination', id: origin.id }}
```
Comment (clarani) [inline front/src/applications/stdcm/components/StdcmForm/StdcmConfig.tsx:152]: > [!NOTE]
Actually, we can only give the extremityType and all the other args can be deduced from it.
We can do this in a future PR

**Review by clarani — APPROVED** [review]
LGTM overall :white_check_mark: 

I created an enhancement issue for this feature: https://github.com/OpenRailAssociation/osrd/issues/9545.
We should do it as soon as we can (next week)

File: front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx (hunk 28 above)
Comment (clarani) [reply front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx]: We can do this later :+1:

File: front/src/styles/scss/applications/stdcm/_linkedPath.scss (hunk 21 above)
Comment (clarani) [reply front/src/styles/scss/applications/stdcm/_linkedPath.scss]: I close this discussion :+1:
